### ISSUE TITLE
error - no error ingame / console

### ISSUE THREAD

**Issue by hasezoey** [mrmakeit/OCJS#32]
i only try to get errors ingame / console nad then this comes
```
13:27:25] [ForkJoinPool-1-worker-3/INFO] [opencomputers]: Running the latest OpenComputers version.
[13:28:40] [Server thread/WARN] [opencomputers]: Failed initializing computer.
java.lang.NoClassDefFoundError: org/slf4j/LoggerFactory
        at delight.nashornsandbox.internal.NashornSandboxImpl.<clinit>(NashornSandboxImpl.java:38) ~[NashornSandboxImpl.class:?]
        at delight.nashornsandbox.NashornSandboxes.create(NashornSandboxes.java:24) ~[NashornSandboxes.class:?]
        at me.mrmakeit.ocjs.NashornAPI.<init>(NashornAPI.java:33) ~[NashornAPI.class:?]
        at me.mrmakeit.ocjs.NashornBabelArch.initialize(NashornBabelArch.java:28) ~[NashornBabelArch.class:?]
        at li.cil.oc.server.machine.Machine.init(Machine.scala:884) [Machine.class:?]
        at li.cil.oc.server.machine.Machine.start(Machine.scala:228) [Machine.class:?]
        at li.cil.oc.server.PacketHandler$.trySetComputerPower(PacketHandler.scala:118) [PacketHandler$.class:?]
        at li.cil.oc.server.PacketHandler$.onComputerPower(PacketHandler.scala:61) [PacketHandler$.class:?]
        at li.cil.oc.server.PacketHandler$.dispatch(PacketHandler.scala:36) [PacketHandler$.class:?]
        at li.cil.oc.common.PacketHandler.li$cil$oc$common$PacketHandler$$process(PacketHandler.scala:53) [PacketHandler.class:?]
        at li.cil.oc.common.PacketHandler$$anon$1.run(PacketHandler.scala:40) [PacketHandler$$anon$1.class:?]
        at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) [?:1.8.0_172]
        at java.util.concurrent.FutureTask.run(Unknown Source) [?:1.8.0_172]
        at net.minecraft.util.Util.func_181617_a(SourceFile:46) [h.class:?]
        at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:723) [MinecraftServer.class:?]
        at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:396) [nz.class:?]
        at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:668) [MinecraftServer.class:?]
        at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:526) [MinecraftServer.class:?]
        at java.lang.Thread.run(Unknown Source) [?:1.8.0_172]
Caused by: java.lang.ClassNotFoundException: org.slf4j.LoggerFactory
        at net.minecraft.launchwrapper.LaunchClassLoader.findClass(LaunchClassLoader.java:101) ~[launchwrapper-1.12.jar:?]
        at java.lang.ClassLoader.loadClass(Unknown Source) ~[?:1.8.0_172]
        at java.lang.ClassLoader.loadClass(Unknown Source) ~[?:1.8.0_172]
        ... 19 more
```

**Comment by mrmakeit** [comment]
Can you provide the following items?
1. Minecraft Version:
1. OpenComputers Version:
1. OCJS Version:
1. Full log in the form of a pastebin link
1. If this is happening during JS Code execution, all code that may have been run on the computer.

I don't see why LoggerFactory is being requested and not available, so these should help narrow it down/

**Comment by hasezoey** [comment]
Minecraft Version: 1.12.2
OpenComputers V: 1.7.2.67 (newest at Curse)
OCJS: 0.6.1 (+ self compiling)
[Full Log](https://gist.github.com/hasezoey/3d076b5fd53677b9a7fe8ce3cb926368) (its only a snippet, because it would be too long)

**Comment by mrmakeit** [comment]
Thanks, I'll take a look at this in a bit.

**Comment by mrmakeit** [comment]
Following javadelight/delight-nashorn-sandbox#48 as this appears to be due to the SLF4J dependency not being able to be embedded correctly.

**Comment by hasezoey** [comment]
what means the Label "BLOCKING"?

**Comment by mrmakeit** [comment]
Looks like the issue was that nashorn-sandbox depends on slf4j which doesn't correctly embed in minecraft mods, so I've rebuilt https://github.com/mrmakeit/delight-nashorn-sandbox without it.  Seems to fix the issue.  Try https://github.com/mrmakeit/OCJS/releases/download/0.6.1/ocjs-0.6.1-1.12.2.jar

**Comment by hasezoey** [comment]
is that really the new file? seems like the version like i have yet

**Comment by hasezoey** [comment]
yes it works yet ~~with "my" link ~~  

but yet i get strange "TypeError"'s (with the JS EEPROM) 
https://gist.github.com/hasezoey/0781a941c591e0d37464d776ed2d34c3
(is this a babel error?)

**Comment by mrmakeit** [comment]
Looks like I pasted the wrong link. The one you have is correct.
Make sure you refresh the eeprom by getting a new one from the creative menu.  EEPROM code is stored in the item's NBT data, and doesn't automatically update when bios.js is changed.

**Comment by hasezoey** [comment]
still the same error

**Comment by mrmakeit** [comment]
hmm.  can you provide the init.js file that you are using?

**Comment by hasezoey** [comment]
i use "EEPROM (JS BIOS)"

**Comment by mrmakeit** [comment]
no floppy disk or harddrive?

**Comment by hasezoey** [comment]
uhm no only the eeprom

**Comment by mrmakeit** [comment]
hu.  Can you get me the entire log file?

**Comment by hasezoey** [comment]
here: https://gist.github.com/hasezoey/0f3099b81954500e501287eb404b7b38

**Comment by mrmakeit** [comment]
It appears that maybe your on the lua architecture?  
If not, I can't seem to reproduce on my setup, so a world file would be very helpful.

**Comment by mrmakeit** [comment]
nope, never mind, your on the right architecture.  I'll need a sample world file.

**Comment by hasezoey** [comment]
here is the test modpack (Curse /Twitch Launcher) with the saves

**Comment by mrmakeit** [comment]
Did you forget to paste the link?

**Comment by hasezoey** [comment]
um where i first posted it, yes

**Comment by mrmakeit** [comment]
I think github might have blocked the link.

**Comment by mrmakeit** [comment]
![image](https://user-images.githubusercontent.com/764185/41188817-327ad742-6b89-11e8-8062-8aaf12ea1c16.png)

**Comment by hasezoey** [comment]
also at me the link is still there, reload?

**Comment by mrmakeit** [comment]
hu.  Well, got the link now.

**Comment by mrmakeit** [comment]
Well, I'm not receiving any typeerrors. ES6 mode can obfuscate errors by minimizing the code before execution. Can you try running it in ES5 mode and paste the error that occurs?

**Comment by hasezoey** [comment]
then it only says "No event Loop"

**Comment by mrmakeit** [comment]
That is normal.  If it can't find a init.js entry in any filesystem, it doesn't load an event loop.
I want a better error message, but catching filesystem access failure across runThreaded calls is a bit more complicated.

**Comment by mrmakeit** [comment]
For the sake of completeness, can you try switching back to es6 and see if the previous error comes back?

**Comment by hasezoey** [comment]
wait, when i execute it in ES6 again it says "SyntaxError: empty range in char class in <eval> at line number 1 at column number 515642"

**Comment by mrmakeit** [comment]
The computer goes into infinite sleep after loading the eeprom (can be changed by calling `computer.sleep()`), and only responds when a signal is pushed.

**Comment by hasezoey** [comment]
i dont understand, nowhere "computer.sleep" is executet in the default eeprom?

**Comment by mrmakeit** [comment]
Try clicking the screen after booting it up.
Should immediately shutdown.
computer.sleep changes the sleep timer. It's due to the execution loop requirements of nashorn and rhino.  All JS must completely execute and end the main function before control can be passed back to OpenComputers.  Memory contents (scope) gets preserved between loops, but execution has to stop.
As for the ES6 issue, I'm at a loss right now.  I can't get the issue to show up on my end at all, and looking at the first logs, it seems the issue changed between executions (first the empty range issue, then the type error), neither of which seem to be executing code in the eeprom.  I can't figure out what is going on.

**Comment by mrmakeit** [comment]
Since JavaScript is an event based language, instead of stack based, all code has to complete before returning to the higher level functions (Java).
As such, unlike lua, which can pause execution anywhere in the script, OCJS has to call a function every time `runThreaded` is executed by OC.  `runThreaded` is called either after the sleep timer expires, or when a new signal is pushed.

**Comment by hasezoey** [comment]
(i think this is out of context, but)
(CPU: ES5)
when i start it up it says no event loop (normal)
but the pc is still green (still working)
and when i then pull out the cpu (to change to ES6) it says then on the screen: "Too many components connected to the computer"

**Comment by mrmakeit** [comment]
Yes, pulling the cpu out while it is still running reduces the available component limit to 0
The no event loop is from the previous execution (Screen doesn't clear when the computer boots up, so the previous error is left up)

**Comment by mrmakeit** [comment]
I do want to continue helping with this issue, but since the original issue is resolved, I'm going to close this issue.  If you're still having issues with ES6 mode, please do create another issue with the appropriate info.  It helps others searching for the same issue.

**Comment by hasezoey** [comment]
i try to make a small custom EEPROM and the TypeError (the exact same) is there again...
